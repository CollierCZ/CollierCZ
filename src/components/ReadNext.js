import { css } from '@emotion/core'
import React, { Component } from "react"
import classNames from "classnames"
import Link from "gatsby-link"

const ReadNextStory = props => {
  const { article, text } = props;
  if (article) {
    const { path, cover, title, excerpt } = article;
    const classes = classNames("read-next-story", props.className);
    return (
      <Link className={classes} 
        to={path}
        css={css`
          display: -webkit-box;
          display: -webkit-flex;
          display: -ms-flexbox;
          display: flex;
          -webkit-box-flex: 1;
          -webkit-flex-grow: 1;
          -ms-flex-positive: 1;
          flex-grow: 1;
          min-width: 50%;
          text-decoration: none;
          position: relative;
          text-align: center;
          color: #fff;
          background: #222 no-repeat center center;
          background-size: cover;
          overflow: hidden;
          ${cover ? `background-image: url(${cover});` : null}
          :hover::before {
            background: rgba(0, 0, 0, 0.8);
            transition: all 0.2s ease;
          }
          :hover article::before {
            color: #222;
            background: #fff;
            transition: all 0.2s ease;
          }
          /*darken the image */
          ::before {
            content: "";
            display: block;
            position: absolute;
            top: 0;
            right: 0;
            bottom: 0;
            left: 0;
            background: rgba(0, 0, 0, 0.7);
            transition: all 0.5s ease;
        }
        `}
      >
        <article 
          css={css`
            position: relative;
            width: 80%;
            max-width: 710px;
            margin: 4rem auto;
            padding-bottom: 4rem;
            word-wrap: break-word;
            @media only screen and (max-width: 500px) {
              font-size: 0.95em;
            }
            @media only screen and (max-width: 500px) {
                width: 100%;
                margin: 2rem 16px;
                padding-bottom: 2rem;
                line-height: 1.65em;
            }
            ::before {
              content: "${text}";
              padding: 4px 10px 5px;
              text-transform: uppercase;
              font-size: 1.1rem;
              font-family: "Open Sans", sans-serif;
              color: rgba(255, 255, 255, 0.8);
              border: rgba(255, 255, 255, 0.5) 1px solid;
              border-radius: 4px;
              transition: all 0.5s ease;
            }
          `}
        >
          <h2
            css={css`
              margin-top: 1rem;
              color: #fff;
            `}
          >{title}</h2>
          <p
            css={css`
              margin: 0;
              color: rgba(255, 255, 255 0.8);
              @media only screen and (max-width: 500px) {
                display:none;
              }
            `}
          >{excerpt}&hellip;</p>
        </article>
      </Link>
    );
  }
  return null;
};

/**
 * Links to Previous/Next articles
 */
class ReadNext extends Component {
  render() {
    const { next, prev } = this.props;
    return (
      <aside
        css={css`
          display: -webkit-box;
          display: -webkit-flex;
          display: -ms-flexbox;
          display: flex;
          -webkit-box-align: stretch;
          -webkit-align-items: stretch;
          -ms-flex-align: stretch;
          align-items: stretch;
          margin-top: 10rem;
          @media only screen and (max-width: 900px) {
            -webkit-box-orient: vertical;
            -webkit-box-direction: normal;
            -webkit-flex-direction: column;
            -ms-flex-direction: column;
            flex-direction: column;
            margin-top: 4rem;
          }
          @media only screen and (max-width: 500px) {
            margin-top: 2rem;
            margin-bottom: 0.5rem;
          }
        `}
      >
        <ReadNextStory text="Read This Next" article={next} />
        <ReadNextStory text="You Might Enjoy" article={prev} />
      </aside>
    );
  }
}

export default ReadNext;
