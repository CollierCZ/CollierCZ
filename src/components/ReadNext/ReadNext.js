import React, { Component } from "react";
import classNames from "classnames";
import Link from "gatsby-link";
import "./ReadNext.css";

const ReadNextStory = props => {
  const { article } = props;
  if (article) {
    /** Limit words to 19 in GraphQL * */
    const { path, cover, title, excerpt } = article;
    const styles = cover ? { backgroundImage: `url(${cover}` } : null;
    const classes = classNames("read-next-story", props.className, {
      "no-cover": !cover
    });
    return (
      <Link className={classes} style={styles} to={path}>
        <section className="article">
          <h2>{title}</h2>
          <p>{excerpt}&hellip;</p>
        </section>
      </Link>
    );
  }
  return null;
};

/**
 * Links to Previous/Next articles
 */
class ReadNext extends Component {
  render() {
    const { next, prev } = this.props;
    return (
      <aside className="read-next">
        <ReadNextStory article={next} />
        <ReadNextStory className="prev" article={prev} />
      </aside>
    );
  }
}

export default ReadNext;
