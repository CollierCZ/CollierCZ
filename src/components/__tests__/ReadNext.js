import React from "react"
import { shallow } from "enzyme"
import ReadNext, { ReadNextStory } from "../ReadNext"

const article = {
    title: "How I built and deployed this site for free",
    cover: "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/38bbc989-70ab-4661-92cf-11efafa5ab09/sunset-219354_1920.jpg",
    path: "how-i-built-and-deployed-this-site-for-free",
    excerpt: "How I built a modern static site at no cost using Gatsby and Kentico Kontent for Content as a Service."
}
const articleNoCover = {
  title: "How I built and deployed this site for free",
  path: "how-i-built-and-deployed-this-site-for-free",
  excerpt: "How I built a modern static site at no cost using Gatsby and Kentico Kontent for Content as a Service."
}

describe("ReadNext", () => {
  it("renders correctly with articles with and without covers", () => {
    const component = shallow(<ReadNext next={article} prev={articleNoCover} />)
    expect(component).toMatchSnapshot()
  })
})

describe("ReadNextStory", () => {
  it("renders correctly with articles with cover", () => {
    const component = shallow(<ReadNextStory article={article} text="Read This Next" />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with articles with no cover", () => {
    const component = shallow(<ReadNextStory article={articleNoCover} text="Read This Next" />)
    expect(component).toMatchSnapshot()
  })
})