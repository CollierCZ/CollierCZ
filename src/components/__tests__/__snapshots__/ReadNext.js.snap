// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadNext renders correctly with articles with and without covers 1`] = `
.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 16px;
  font-weight: 500;
  color: #FFFFFF;
  line-height: 24px;
  margin: 0;
  margin-bottom: 8px;
}

.c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 28px;
  font-weight: 700;
  color: #FFFFFF;
  line-height: 36px;
  margin: 0;
  margin-bottom: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c0 > * {
  margin: 0 0 16px 0!important;
}

.c0 > *:last-child {
  margin: 0 !important;
}

.c3 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #FFFFFF;
  line-height: 20px;
  text-align: center;
  margin: 0;
}

.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

@media (min-width:414px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {
  .c0 > * {
    margin: 0 0 16px 0!important;
  }
}

@media (min-width:1200px) {

}

<aside
  css={
    Array [
      "
        display: flex;
        align-items: stretch;
        margin-top: 32px;
        @media only screen and (max-width: 900px) {
          flex-direction: column;
        }
      ",
    ]
  }
>
  <a
    className="read-next-story"
    css={
      Array [
        "
        display: flex;
        flex-grow: 1;
        min-width: 50%;
        text-decoration: none;
        position: relative;
        background: #222 no-repeat center center;
        background-size: cover;
        overflow: hidden;
        ",
        "background-image: url(https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/38bbc989-70ab-4661-92cf-11efafa5ab09/sunset-219354_1920.jpg&w=1200);",
        " :hover::before {
          background: rgba(0, 0, 0, 0.8);
          transition: all 0.2s ease;
        }

        /*darken the image */
        ::before {
          content: \\"\\";
          display: block;
          position: absolute;
          top: 0;
          right: 0;
          bottom: 0;
          left: 0;
          background: rgba(0, 0, 0, 0.7);
          transition: all 0.5s ease;
        }
      ",
      ]
    }
    href="/how-i-built-and-deployed-this-site-for-free"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <article
      css={
        Array [
          "
          position: relative;
          width: 80%;
          max-width: 710px;
          margin: 52px auto;
        ",
        ]
      }
    >
      <div
        className="c0"
      >
        <div
          className="c1"
        >
          Read This Next
        </div>
        <div
          className="c2"
        >
          How I built and deployed this site for free
        </div>
        <p
          className="c3"
        >
          How I built a modern static site at no cost using Gatsby and Kentico Kontent for Content as a Service.
        </p>
      </div>
    </article>
  </a>
  <a
    className="read-next-story"
    css={
      Array [
        "
        display: flex;
        flex-grow: 1;
        min-width: 50%;
        text-decoration: none;
        position: relative;
        background: #222 no-repeat center center;
        background-size: cover;
        overflow: hidden;
        ",
        " :hover::before {
          background: rgba(0, 0, 0, 0.8);
          transition: all 0.2s ease;
        }

        /*darken the image */
        ::before {
          content: \\"\\";
          display: block;
          position: absolute;
          top: 0;
          right: 0;
          bottom: 0;
          left: 0;
          background: rgba(0, 0, 0, 0.7);
          transition: all 0.5s ease;
        }
      ",
      ]
    }
    href="/how-i-built-and-deployed-this-site-for-free"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <article
      css={
        Array [
          "
          position: relative;
          width: 80%;
          max-width: 710px;
          margin: 52px auto;
        ",
        ]
      }
    >
      <div
        className="c0"
      >
        <div
          className="c1"
        >
          You Might Enjoy
        </div>
        <div
          className="c2"
        >
          How I built and deployed this site for free
        </div>
        <p
          className="c3"
        >
          How I built a modern static site at no cost using Gatsby and Kentico Kontent for Content as a Service.
        </p>
      </div>
    </article>
  </a>
</aside>
`;

exports[`ReadNextStory renders correctly with articles with cover 1`] = `
.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 16px;
  font-weight: 500;
  color: #FFFFFF;
  line-height: 24px;
  margin: 0;
  margin-bottom: 8px;
}

.c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 28px;
  font-weight: 700;
  color: #FFFFFF;
  line-height: 36px;
  margin: 0;
  margin-bottom: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c0 > * {
  margin: 0 0 16px 0!important;
}

.c0 > *:last-child {
  margin: 0 !important;
}

.c3 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #FFFFFF;
  line-height: 20px;
  text-align: center;
  margin: 0;
}

.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

@media (min-width:414px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {
  .c0 > * {
    margin: 0 0 16px 0!important;
  }
}

@media (min-width:1200px) {

}

<a
  className="read-next-story"
  css={
    Array [
      "
        display: flex;
        flex-grow: 1;
        min-width: 50%;
        text-decoration: none;
        position: relative;
        background: #222 no-repeat center center;
        background-size: cover;
        overflow: hidden;
        ",
      "background-image: url(https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/38bbc989-70ab-4661-92cf-11efafa5ab09/sunset-219354_1920.jpg&w=1200);",
      " :hover::before {
          background: rgba(0, 0, 0, 0.8);
          transition: all 0.2s ease;
        }

        /*darken the image */
        ::before {
          content: \\"\\";
          display: block;
          position: absolute;
          top: 0;
          right: 0;
          bottom: 0;
          left: 0;
          background: rgba(0, 0, 0, 0.7);
          transition: all 0.5s ease;
        }
      ",
    ]
  }
  href="/how-i-built-and-deployed-this-site-for-free"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <article
    css={
      Array [
        "
          position: relative;
          width: 80%;
          max-width: 710px;
          margin: 52px auto;
        ",
      ]
    }
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Read This Next
      </div>
      <div
        className="c2"
      >
        How I built and deployed this site for free
      </div>
      <p
        className="c3"
      >
        How I built a modern static site at no cost using Gatsby and Kentico Kontent for Content as a Service.
      </p>
    </div>
  </article>
</a>
`;

exports[`ReadNextStory renders correctly with articles with no cover 1`] = `
.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 16px;
  font-weight: 500;
  color: #FFFFFF;
  line-height: 24px;
  margin: 0;
  margin-bottom: 8px;
}

.c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 28px;
  font-weight: 700;
  color: #FFFFFF;
  line-height: 36px;
  margin: 0;
  margin-bottom: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c0 > * {
  margin: 0 0 16px 0!important;
}

.c0 > *:last-child {
  margin: 0 !important;
}

.c3 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #FFFFFF;
  line-height: 20px;
  text-align: center;
  margin: 0;
}

.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c3 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

@media (min-width:414px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {
  .c0 > * {
    margin: 0 0 16px 0!important;
  }
}

@media (min-width:1200px) {

}

<a
  className="read-next-story"
  css={
    Array [
      "
        display: flex;
        flex-grow: 1;
        min-width: 50%;
        text-decoration: none;
        position: relative;
        background: #222 no-repeat center center;
        background-size: cover;
        overflow: hidden;
        ",
      " :hover::before {
          background: rgba(0, 0, 0, 0.8);
          transition: all 0.2s ease;
        }

        /*darken the image */
        ::before {
          content: \\"\\";
          display: block;
          position: absolute;
          top: 0;
          right: 0;
          bottom: 0;
          left: 0;
          background: rgba(0, 0, 0, 0.7);
          transition: all 0.5s ease;
        }
      ",
    ]
  }
  href="/how-i-built-and-deployed-this-site-for-free"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <article
    css={
      Array [
        "
          position: relative;
          width: 80%;
          max-width: 710px;
          margin: 52px auto;
        ",
      ]
    }
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Read This Next
      </div>
      <div
        className="c2"
      >
        How I built and deployed this site for free
      </div>
      <p
        className="c3"
      >
        How I built a modern static site at no cost using Gatsby and Kentico Kontent for Content as a Service.
      </p>
    </div>
  </article>
</a>
`;
